### ISSUE TITLE
About loss

### ISSUE THREAD

**Issue by arufus** [ifzhang/FairMOT#249]
@ifzhang  Thank you for your amazing work!  Sorry to bother again.

can you explain more about the loss?:

> https://github.com/ifzhang/FairMOT/blob/master/src/lib/trains/mot.py
```python
loss = torch.exp(-self.s_det) * det_loss + torch.exp(-self.s_id) * id_loss + (self.s_det + self.s_id)
```
Why the total loss need to add `(self.s_det + self.s_id)` ?  and this is a negative value, it may cause the total loss to be a negative value too (when using my own data), I am so confused and didn't find a explain in paper. Thank you!

**Comment by ifzhang** [comment]
You can find more details in the paper "Multi-task learning using uncertainty to weigh losses for scene geometry and semantics" https://arxiv.org/abs/1705.07115. It is an uncertainty loss that automaticly balance different tasks.

**Comment by arufus** [comment]
Thanks for your answer !

**Comment by zws510** [comment]
> Thanks for your answer !

How did you solve the problem of negative loss

**Comment by zws510** [comment]
> Thanks for your answer !

How did you solve the problem of negative loss? Thanks.

**Comment by long2double** [comment]
Hello,

self.s_det = nn.Parameter(-1.85 * torch.ones(1))
self.s_id = nn.Parameter(-1.05 * torch.ones(1))

How is the initial value of this constant determined?

**Comment by HanGuangXin** [comment]
> Hello,
> 
> self.s_det = nn.Parameter(-1.85 * torch.ones(1)) self.s_id = nn.Parameter(-1.05 * torch.ones(1))
> 
> How is the initial value of this constant determined?

Randomly selected, as the owner said.
